lib/template.html

```diff
@@ -111,27 +93,56 @@
   {% endif %}
 {% endif %}
 
+<script type="application/ld+json">
+  {
+    "@context" : "http://schema.org",
+
+{% if page.seo and page.seo.type %}
+    "@type" : {{ page.seo.type | jsonify }},
+{% elsif page.url == "/" %}
+    "@type" : "WebSite",
+{% elsif page.id %}
```

Comment:
I'd check if it has a date? Even if it's a collection with a date (possible in 3.0), it'd be closer to a post (dated, chronological), than a page (unordered, undated).